drivers/mspi/mspi_nrfe.c

```diff
@@ -0,0 +1,834 @@
+}
+
+static void ep_recv(const void *data, size_t len, void *priv)
+{
+	(void)priv;
+
+	ipc_recv_clbk(data, len);
+}
+
+static struct ipc_ept_cfg ep_cfg = {
+	.cb = {
+		.bound = ep_bound,
+		.received = ep_recv,
+	},
+};
+
+/**
+ * @brief IPC receive callback function.
+ *
+ * This function is called by the IPC stack when a message is received from the
+ * other core. The function checks the opcode of the received message and takes
+ * appropriate action.
+ *
+ * @param data Pointer to the received message.
+ * @param len Length of the received message.
+ */
+static void ipc_recv_clbk(const void *data, size_t len)
+{
+	nrfe_mspi_flpr_response_t *response = (nrfe_mspi_flpr_response_t *)data;
+
+	switch (response->opcode) {
+	case NRFE_MSPI_CONFIG_PINS: {
+#if defined(CONFIG_MULTITHREADING)
+		k_sem_give(&ipc_sem_cfg);
+#else
+		atomic_set_bit(&ipc_atomic_sem, NRFE_MSPI_CONFIG_PINS);
+#endif
+		break;
+	}
+	case NRFE_MSPI_CONFIG_CTRL: {
+#if defined(CONFIG_MULTITHREADING)
+		k_sem_give(&ipc_sem_cfg);
+#else
+		atomic_set_bit(&ipc_atomic_sem, NRFE_MSPI_CONFIG_CTRL);
+#endif
+		break;
+	}
+	case NRFE_MSPI_CONFIG_DEV: {
+#if defined(CONFIG_MULTITHREADING)
+		k_sem_give(&ipc_sem_cfg);
+#else
+		atomic_set_bit(&ipc_atomic_sem, NRFE_MSPI_CONFIG_DEV);
+#endif
+		break;
+	}
+	case NRFE_MSPI_CONFIG_XFER: {
+#if defined(CONFIG_MULTITHREADING)
+		k_sem_give(&ipc_sem_cfg);
+#else
+		atomic_set_bit(&ipc_atomic_sem, NRFE_MSPI_CONFIG_XFER);
+#endif
+		break;
+	}
+	case NRFE_MSPI_TX: {
+#if defined(CONFIG_MULTITHREADING)
+		k_sem_give(&ipc_sem_xfer);
+#else
+		atomic_set_bit(&ipc_atomic_sem, NRFE_MSPI_TX);
+#endif
+		break;
+	}
+	case NRFE_MSPI_TXRX: {
+		if (len) {
+			ipc_received = len - 1;
+			ipc_receive_buffer = (uint8_t *)&response->data;
+		}
+#if defined(CONFIG_MULTITHREADING)
+		k_sem_give(&ipc_sem_xfer);
+#else
+		atomic_set_bit(&ipc_atomic_sem, NRFE_MSPI_TXRX);
+#endif
+		break;
+	}
+	default: {
+		LOG_ERR("Invalid response opcode: %d", response->opcode);
+		break;
+	}
+	}
+
+	LOG_DBG("Received msg with opcode: %d", response->opcode);
+}
+
+/**
+ * @brief Send data to the flpr with the given opcode.
+ *
+ * @param opcode The opcode of the message to send.
+ * @param data The data to send.
+ * @param len The length of the data to send.
+ *
+ * @return 0 on success, -ENOMEM if there is no space in the buffer,
+ *         -ETIMEDOUT if the transfer timed out.
+ */
+static int mspi_ipc_data_send(enum nrfe_mspi_opcode opcode, const void *data, size_t len)
+{
+	int rc;
+
+	LOG_DBG("Sending msg with opcode: %d", (uint8_t)opcode);
+#if defined(CONFIG_SYS_CLOCK_EXISTS)
+	uint32_t start = k_uptime_get_32();
+#else
+	uint32_t repeat = EP_SEND_TIMEOUT_MS;
+#endif
+#if !defined(CONFIG_MULTITHREADING)
+	atomic_clear_bit(&ipc_atomic_sem, opcode);
+#endif
+
+	do {
+		rc = ipc_service_send(&ep, data, len);
+#if defined(CONFIG_SYS_CLOCK_EXISTS)
+		if ((k_uptime_get_32() - start) > EP_SEND_TIMEOUT_MS) {
+#else
+		repeat--;
+		if ((rc < 0) && (!repeat)) {
```

Comment:
```suggestion
		if ((rc < 0) && (repeat != 0)) {
```
Please, do not use logic operators on numbers, it's less readable.